webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/ProjectExportService.scala

```diff
@@ -0,0 +1,209 @@
+package org.knora.webapi.slice.admin.domain.service
+
+import zio.Task
+import zio.ZIO
+import zio.macros.accessible
+
+import java.io.BufferedInputStream
+import java.io.BufferedOutputStream
+import java.nio.file.Files
+import java.nio.file.Path
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
+
+import org.knora.webapi.IRI
+import org.knora.webapi.core.MessageRelay
+import org.knora.webapi.messages.StringFormatter
+import org.knora.webapi.messages.store.triplestoremessages.FileWrittenResponse
+import org.knora.webapi.messages.store.triplestoremessages.NamedGraphFileRequest
+import org.knora.webapi.messages.twirl
+import org.knora.webapi.messages.util.rdf.RdfFeatureFactory
+import org.knora.webapi.messages.util.rdf.RdfFormatUtil
+import org.knora.webapi.messages.util.rdf.RdfInputStreamSource
+import org.knora.webapi.messages.util.rdf.RdfStreamProcessor
+import org.knora.webapi.messages.util.rdf.Statement
+import org.knora.webapi.messages.util.rdf.TriG
+import org.knora.webapi.slice.admin.AdminConstants.adminDataGraph
+import org.knora.webapi.slice.admin.AdminConstants.permissionsDataGraph
+import org.knora.webapi.slice.admin.domain.model.KnoraProject
+import org.knora.webapi.slice.ontology.domain.service.OntologyRepo
+import org.knora.webapi.store.triplestore.api.TriplestoreService
+
+@accessible
+trait ProjectExportService {
+
+  /**
+   * Exports a project to a file.
+   * The file format is TriG.
+   * The data exported is:
+   *  * the project metadata
+   *  * the project's permission data
+   *  * the triples of the project's ontologies
+   *
+   * @param project the project to be exported
+   * @param file the path to the file to which the project should be exported
+   * @return the [[Path]] to the file to which the project was exported
+   */
+  def exportProjectTriples(project: KnoraProject, file: Path): Task[Path]
+}
+
+/**
+ * Represents a named graph to be saved to a TriG file.
+ *
+ * @param graphIri the IRI of the named graph.
+ * @param tempDir  the directory in which the file is to be saved.
+ */
+private case class NamedGraphTrigFile(graphIri: IRI, tempDir: Path) {
+  lazy val dataFile: Path = {
+    val filename = graphIri.replaceAll("[.:/]", "_") + ".trig"
+    tempDir.resolve(filename)
+  }
+}
+
+/**
+ * An [[RdfStreamProcessor]] for combining several named graphs into one.
+ *
+ * @param formattingStreamProcessor an [[RdfStreamProcessor]] for writing the combined result.
+ */
```

Comment:
```suggestion
/**
 * A [[RdfStreamProcessor]] for combining several named graphs into one.
 *
 * @param formattingStreamProcessor a [[RdfStreamProcessor]] for writing the combined result.
 */
```

IMO the argument name could be also simplified just to `processor` as it has already expressive type .